pkgs/development/lisp-modules/import/nix.lisp

```diff
@@ -0,0 +1,81 @@
+(defpackage org.lispbuilds.nix/nix
+  (:documentation "Utilities for generating Nix code")
+  (:use :cl)
+  (:import-from :str)
+  (:import-from :ppcre)
+  (:import-from :arrow-macros :->>)
+  (:import-from :org.lispbuilds.nix/util :replace-regexes)
+  (:export
+   :nix-eval
+   :nixify-symbol
+   :system-master
+   :make-pname
+   :*nix-attrs-depth*))
+
+(in-package org.lispbuilds.nix/nix)
+
+;; Path names are alphanumeric and can include the symbols +-._?= and
+;; must not begin with a period.
+(defun make-pname (string)
+  (replace-regexes '("^[.]" "[^a-zA-Z0-9+-._?=]")
+                   '("_" "_")
+                   string))
+
+(defun system-master (system)
+  (first (str:split "/" system)))
+
+;;;; Nix generation
+
+(defun nix-eval (exp)
+  (assert (consp exp))
+  (ecase (car exp)
+    (:string (nix-string (cadr exp)))
+    (:list (apply #'nix-list (rest exp)))
+    (:funcall (apply #'nix-funcall (rest exp)))
+    (:attrs (nix-attrs (cdr exp)))
+    (:merge (apply #'nix-merge (cdr exp)))
+    (:symbol (nix-symbol (cadr exp)))))
+
+(defun nix-string (object)
+  (format nil "\"~a\"" object))
+
+(defun nixify-symbol (string)
+  (flet ((fix-special-chars (str)
+           (replace-regexes '("[+]$" "[+][/]" "[+]" "[.]" "[/]")
+                            '("_plus" "_plus/" "_plus_" "_dot_" "_slash_")
+                            str)))
```

Comment:
I would also replace `_` with `_underscore_` for reversibility.

Reply:
I did it locally.  It affected 363 names. Hmm. I'm not sure if this is good. It results in a name such as `cl-glfw-opengl-nv_underscore_vertex_underscore_buffer_underscore_unified_underscore_memory` which is very long. Do you think we can skip replacing `_` with `_underscore_`?

Reply:
Hm, maybe `__` would be not too bad then?

Reply:
OK, added [57563cf](https://github.com/NixOS/nixpkgs/pull/218870/commits/57563cf8f0126169809259124cf905c429820f22)

Reply:
Thanks!